connectors/protocol/connectors.py

```diff
@@ -604,7 +604,7 @@ async def heartbeat(self, interval):
             self.log_debug("Sending heartbeat")
             await self.index.heartbeat(doc_id=self.id)
 
-    def next_sync(self, job_type):
+    def next_sync(self, job_type, now):
```

Comment:
This is a major change - before "now" was arbitrary - connector wakes up at 11:59:59, but when checks the schedule it's already 12:00:01 and the moment of scheduling is lost